commcare_connect/opportunity/views.py

```diff
@@ -1376,3 +1363,308 @@ def sync_deliver_units(request, org_slug, opp_id):
+                    "count": user_visit_counts.get("revalidate", 0),
+                },
+                {
+                    "name": "agree",
+                    "label": "Agree",
+                    "count": user_visit_counts.get("approved", 0),
+                },
+                {"name": "all", "label": "All", "count": user_visit_counts.get("total", 0)},
+            ]
+        context = super().get_context_data(**kwargs)
+        context["opportunity_access"] = self.opportunity_access
+        context["tabs"] = tabs
+        return context
+
+    def get_queryset(self):
+        self.opportunity = get_opportunity_or_404(self.kwargs["opp_id"], self.kwargs["org_slug"])
+        self.opportunity_access = get_object_or_404(
+            OpportunityAccess, opportunity=self.opportunity, pk=self.kwargs["pk"]
+        )
+        self.is_program_manager = self.request.org.program_manager and (
+            (self.request.org_membership != None and self.request.org_membership.is_admin)  # noqa: E711
+            or self.request.user.is_superuser
+        )
+
+        self.filter_status = self.request.GET.get("filter_status")
+        self.filter_date = self.request.GET.get("filter_date")
+        filter_kwargs = {"opportunity_access": self.opportunity_access}
+        if self.filter_date:
+            self.filter_date = datetime.datetime.strptime(self.filter_date, "%Y-%m-%d")
+            filter_kwargs.update({"visit_date__date": self.filter_date})
+
+        if self.filter_status == "pending":
+            filter_kwargs.update({"status": VisitValidationStatus.pending})
+            self.exclude_columns = ["last_activity"]
+        if self.filter_status == "approved":
+            filter_kwargs.update({"status": VisitValidationStatus.approved})
+            if self.opportunity.managed:
+                filter_kwargs.update({"review_status": VisitReviewStatus.agree})
+        if self.filter_status == "rejected":
+            filter_kwargs.update({"status": VisitValidationStatus.rejected})
+
+        if self.filter_status == "pending_review":
+            filter_kwargs.update(
+                {
+                    "review_status": VisitReviewStatus.pending,
+                    "status": VisitValidationStatus.approved,
+                }
+            )
+        if self.filter_status == "disagree":
+            filter_kwargs.update(
+                {
+                    "review_status": VisitReviewStatus.disagree,
+                    "status": VisitValidationStatus.approved,
+                }
+            )
+        if self.filter_status == "agree":
+            filter_kwargs.update(
+                {
+                    "review_status": VisitReviewStatus.agree,
+                    "status": VisitValidationStatus.approved,
+                    "flagged": True,
+                }
+            )
+
+        if self.is_program_manager:
+            filter_kwargs.update({"review_created_on__isnull": False})
```

Comment:
Is there a way for program managers to see visits not under review

Reply:
Fixed. [e7114ce](https://github.com/dimagi/commcare-connect/pull/571/commits/e7114ce7e97982e447fb0a970e9c383f23cfda8b)